File: lat2_PHP/memory.php
<?php

class memory extends hardware{

	private $frequency = "";
	private $memorySize = "";
	private $suppCuda = "";

	public function __construct($frequency = "", $memorySize = "", $suppCuda = ""){
		$this->frequency = $frequency;
		$this->memorySize = $memorySize;
		$this->suppCuda = $suppCuda;
	}

	public function setfrequency($frequency){
		$this->frequency = $frequency;
	}

	public function getfrequency(){
		return $this->frequency;
	}

	public function setmemorySize($memorySize){
		$this->memorySize = $memorySize;
	}

	public function getmemorySize(){
		return $this->memorySize;
	}

	public function setsuppCuda($suppCuda){
		$this->suppCuda = $suppCuda;
	}

	public function getsuppCuda(){
		return $this->suppCuda;
	}

	public function printHardware(){
		echo "frequency 		: ". getfrequency() ."<br>";
		echo "Memory Size 		: ". getmemorySize() ."<br>";
		echo "Cuda 		: ". getsuppCuda() ."<br>";
	}
}

File: lat2_python/memory.py
class memory(hardware):

    __frequency = ""
    __memorySize = ""
    __suppCuda = ""

    def __init__(self, frequency = "", memorySize = "", suppCuda = ""):
        self.__frequency = frequency
        self.__memorySize = memorySize
        self.__suppCuda = suppCuda

    def setfrequency(self, frequency):
        self.__frequency = frequency

    def setmemorySize(self, memorySize):
        self.__memorySize = memorySize

    def setsuppCuda(self, suppCuda):
        self.__suppCuda = suppCuda

    def getfrequency(self):
        return self.__frequency

    def getmemorySize(self):
        return self.__memorySize

    def getsuppCuda(self):
        return self.__suppCuda

    def printMemo(self):
        print("Frequency : ", getfrequency())
        print("Memory : ", getmemorySize())
        print("Cuda : ", getsuppCuda())

